lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -104,7 +104,8 @@ class AwsCompileFunctions {
       const functionObject = this.serverless.service.getFunction(functionLogicalName);
 
       newFunction.Properties.Code
-        .S3Key = this.serverless.service.package.artifact.split(path.sep).pop();
+        .S3Key = `${this.serverless.service.package.artifactDirectoryName
+        }/${this.serverless.service.package.artifact.split(path.sep).pop()}`;
```

Comment:
My bad it actually will work as we put it into separate folders that have a date in the folder name, so everything will change. I this case this is fine for me like this